ssm_jax/hmm/learning.py

```diff
@@ -1,37 +1,49 @@
 # Code for parameter estimation (MLE, MAP) using EM and SGD
 
+from functools import partial
+
 import jax.numpy as jnp
-from jax import jit, value_and_grad, vmap
+import jax.random as jr
 import optax
-
+from jax import jit
+from jax import lax
+from jax import value_and_grad
+from jax import vmap
 from tqdm.auto import trange
 
 
-def hmm_fit_em(hmm, batch_emissions, num_iters=50):
+def hmm_fit_em(hmm, batch_emissions, optimizer=optax.adam(1e-2), num_iters=50):
```

Comment:
Because `hmm_fit_em` accepts any HMM model we have and `BaseHMM` does sgd update in its maximisation step, I had to add it even if we don't use the optimizer for the M step of both `GaussianHMM` and `CategoricalHMM`.